### PR TITLE
docs(README): Apply minor updates

### PR DESCRIPTION
**Pull request by krancour**
These are minor doc updates that mainly address name changes among related products-- notably that "Deis v2" is now "Deis Workflow" (should have addressed this a while ago) and that the original "Helm" is now "Helm Classic" (with its binary also renamed from `helm` to `helmc`).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,30 +28,29 @@ The Deis project welcomes contributions from all developers. The high level proc
...
 ```
 Where `<chart>` is selected from the options below:
 
 | Chart | Description |
 |-------|-------------|
-| deis | Install the latest router release along with the rest of the latest Deis platform release. |
-| deis-dev | Install the edge router (from master) with the rest of the edge Deis platform. |
-| router | Install the latest router release with its minimal set of dependencies. |
-| router-dev | Install the edge router (from master) with its minimal set of dependencies. |
+| workflow-beta4 | Install the latest router release along with the rest of the latest Deis Workflow platform release. |
```
Comment (arschles) [inline README.md]: maybe say "latest _stable_ router release"

File: README.md [hunk 2]
```diff
@@ -469,19 +468,19 @@ user agent (browser) --> front-facing load balancer --> a Deis router pod --> ku
...
+* __If using SSL, generate and provide your own dhparam.__  A dhparam is a secret key used in [Diffie Hellman key exchange](https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange) during the SSL handshake in order to help ensure [perfect forward secrecy](https://en.wikipedia.org/wiki/Forward_secrecy).  The Helm Classic charts available for installing router (either with or without the rest of Deis Workflow) already include a dhparam, but recall that dhparams are intended to be secret.  The dhparam included in the charts is marginally preferable to using Nginx's default dhparam only because it is lesser-known, but it is _still_ publicly available in the [deis/charts](https://github.com/deis/charts) repository.  As such, users wishing to run the router in production _and_ use SSL are best off generating their own dhparam.  After being generated, it should be base64 encoded and included as the value of the `dhparam` key in a Kubernetes secret named `deis-router-dhparam` in the same namespace as the router itself.
```
Comment (arschles) [inline README.md]: is it worth finding and linking to a tutorial for creating the `dhparam` secret? I've used http://h71000.www7.hp.com/doc/83final/ba554_90007/ch06s06.html before, but there are probably better ones...

**Comment by arschles** [comment]
@krancour left 2 comments that I don't consider blocking to this PR. I'm marking LGTM on this

File: README.md (hunk 1 above)
Comment (krancour) [inline README.md]: Yes.  Will fix.

File: README.md (hunk 2 above)
Comment (krancour) [inline README.md]: Not a bad thought, but feels separate from the intent of this PR.  I can open a separate issue.

**Pull request merged**